### ISSUE TITLE
Java Backend error for Java interface + Ceylon syntax error

### ISSUE THREAD

**Issue by sgalles** [ceylon/ceylon-ide-eclipse#1249]
Create a Ceylon JMV project, create a module in this project.

In the module create an empty Java interface
```java
public interface Dummy {
}
```
In the module add this ceylon code with a syntax error
```ceylon
shared void run(){  
    blabla
} 
```
*clean* the project you get a bunch of expected errors and also 
```
Some classes are missing from the generated module archives, probably because of an error in the Java backend compilation.
The detail of missing classes is given in the Information markers.
```

**Comment by gavinking** [comment]
@sgalles I can't reproduce this. I get 3 errors from the typechecker. I briefly see that message, but then it gets cleared after a fraction of a second.

**Comment by sgalles** [comment]
@gavinking That's strange. I've recompiled my distrib just to be sure, and after a *clean* in the projet I get the backend error.
Could it be an env difference ? I'm on Windows with Eclipse Kepler...

**Comment by davidfestal** [comment]
@sgalles : this message is expected inside the IDE.

Let me explain : when some error exist in the code, it might be that the Ceylon compiler (backend) omit the generation of some valid classes. The most is done to try generating as much classes as possible, but Javac has limitations in this regard, and sometimes some valid classes may be omitted in the generated archive (please ask to the backend team for me details on this).

Since the Eclipse Build allows incremental build, and by default assumes that a declaration with no syntax or typecheck error *always* generates the related class file, the Ceylon Eclipse Plugin has to track down potential omitted class files inside the archives generated by the backend, in order to include the corresponding source files inside the next incremental build.

The result of this check is given in the message you received, and the detail of the omitted class files is given as information markers, for debugging purposes.  

To summarize, it's not a bug, it's just an additional feature that gives more details about the validity of the generated archives, than the CLI itself gives.

**Comment by davidfestal** [comment]
@sgalles , @gavinking : Are you OK to close this issue ? it's related to a problem we already discussed thoroughly, and this solution was adopted as being sufficiently precise and discreet at the same time.

**Comment by sgalles** [comment]
@davidfestal I missed this discussion and I didn't know this message. So, of course, if there had been a discussion about this behavior it is not a bug and the issue can be closed (still, I must admit that the term **probably** in an error message tends to make me nervous :fearful:  )

**Comment by gavinking** [comment]
It's OK to close this, right @davidfestal?

**Comment by davidfestal** [comment]
yes
